### PR TITLE
Add RangefinderPoint RFC

### PR DESCRIPTION
**Pull request by ojura**
[Rendered](https://github.com/googlecartographer/rfcs/blob/master/text/0027-rangefinderpoint.md)

### CODE REVIEW

File: text/0000-rangefinderpoint.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+Currently, Cartographer is using `Eigen::Vector3/4` directly in lots of places.
+For example, a rangefinder point cloud is an `std::vector` of `Eigen::Vector3/4`.
+Suppose one wants to extend rangefinder points passing through Cartographer to include some metadata (e.g. laser ID).
+Currently, that's not possible, but if Cartographer consistently used a structure named e.g. `RangefinderPoint` for rangefinder points instead of `Eigen::Vector3`, it would be trivial for the user to extend Cartographer to do this. 
+
+## Approach
+[approach]: #approach
+
+Introduce classes `[Timed]RangefinderPoint` and eplace all uses of `Vector3/4` with the new classes.
```
Comment (pifon2a) [inline text/0000-rangefinderpoint.md]: nit: s/eplace/replace

File: text/0000-rangefinderpoint.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+Define `operator*(Rigid3, RangefinderPoint)` for e.g. tracking <-> local frame transformations.
+
+Take care not to break the data flow by avoiding constructing entirely new points - always use metadata-preserving functions on incoming points.
+
+Libcartographer users could then easily modify `RangefinderPoint` to include their metadata.
+
+## Discussion Points
+[discussion]: #discussion
+
+Inheritance of `Vector3/4` instead of composition?
```
Comment (pifon2a) [inline text/0000-rangefinderpoint.md:32]: Composition!

File: text/0000-rangefinderpoint.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+Class naming?
+
+Proto compatibility?
```
Comment (pifon2a) [inline text/0000-rangefinderpoint.md]: That would be nice.

**Review by pifon2a — APPROVED** [review]

**Comment by ojura** [comment]
@pifon2a you forgot to change the number :)

**Pull request merged**